app/components/oauth/oauth.controller.ts

```diff
@@ -56,15 +62,31 @@ export class OAuthController implements angular.IController {
               // persist the user
               _this.AuthService.setUser(user, token, ttl);
 
-              // redirect to original destination (or default to './')
-              var destination = then || './';
-              if (URI(destination).is('absolute')) {
-                _this.authLogger.log('OAuthController, invalid absolute redirect', destination);
-                destination = './';
+              // Check if this is the first time this user has logged in, if so launch the tour
+              var loggedInKey: string = "hasLoggedIn/" + user.metadata.name;
+              var hasLoggedIn: boolean = localStorage.getItem(loggedInKey) === 'true';
+
+              if (hasLoggedIn) {
+                // redirect to original destination (or default to './')
+                var destination = then || './';
+                if (URI(destination).is('absolute')) {
+                  _this.authLogger.log('OAuthController, invalid absolute redirect', destination);
+                  destination = './';
+                }
+                _this.authLogger.log('OAuthController, redirecting', destination);
+                _this.$location.replace();
+                _this.$location.url(destination);
+              } else {
+                localStorage.setItem(loggedInKey, 'true');
+                _this.$location.replace();
+                _this.$location.url('./');
+                this.$timeout(function() {
+                  let tourConfig = _this.Constants.LANDING_PAGE_TOUR_STEPS;
+                  if (tourConfig && tourConfig.steps) {
+                    _this.GuidedTourService.startTour(tourConfig);
```

Comment:
i know this is the example in the catalog app, but thinking about this from the context of how it will behave in the console I have a few concerns

1) a customer may not want their users forced to go through a guided tour, so we either need to let them turn off the tour completely and hide the button (possibly by null-ing the LANDING_PAGE_TOUR_STEPS, or setting it to an empty array), or we have to let them disable automatically sending a user through the guided tour

2) If someone sends me a link to a particular page like logs for a pod, and I'm logging in for the first time, then I still want that to take me to the log page. So we can't always start the guided tour for someone who logs in for the first time.  We need to check if the "then" is set or not.

3) We should consider upgraded clusters.  Users may have been logged in before on the cluster, but the token would never have been set in localStorage.  As an existing user on a cluster would I be annoyed by a guided tour popping up, even if its clearly for a drastically new experience?  I'm on the fence on this one.  cc @serenamarie125